### PR TITLE
Fix custom data point

### PR DESCRIPTION
**Pull request by tgurriet**
More usefull like this

### CODE REVIEW

**Comment by artem-ogre** [comment]
Thank you for the suggestion!

Originally I had it this way too, but then I thought that there is no use case to have different signatures for getters. Do you have some use case in mind, got any examples?

**Comment by artem-ogre** [comment]
Oh, is it because lambdas can have different types?

**Comment by tgurriet** [comment]
> Oh, is it because lambdas can have different types?

Yes :)

**Comment by artem-ogre** [comment]
Yes, silly of me. Thank you for spotting this :)
Please update doxygen comments and synopsis in README.md and I will gladly merge.

File: CDT/include/CDT.h [hunk 1]
```diff
@@ -72,37 +72,45 @@ class CDT_EXPORT Triangulation
     /**
      * Insert custom point-types specified by iterator range and X/Y-getters
      * @tparam TVertexIter iterator that dereferences to custom point type
-     * @tparam TGetVertexCoord function object getting coordinate from* vertex.
+     * @tparam TGetVertexCoordX function object getting x coordinate from* vertex.
```
Comment (artem-ogre) [inline CDT/include/CDT.h:75]:
```suggestion
     * @tparam TGetVertexCoordX function object getting x coordinate from vertex.
```

File: CDT/include/CDT.h [hunk 2]
```diff
@@ -72,37 +72,45 @@ class CDT_EXPORT Triangulation
...
+     * @tparam TGetVertexCoordY function object getting y coordinate from* vertex.
```
Comment (artem-ogre) [inline CDT/include/CDT.h:76]:
```suggestion
     * @tparam TGetVertexCoordY function object getting y coordinate from vertex.
```

File: CDT/include/CDT.h [hunk 3]
```diff
@@ -72,37 +72,45 @@ class CDT_EXPORT Triangulation
...
-        TGetVertexCoord getY);
+        TGetVertexCoordX getX,
+        TGetVertexCoordY getY);
     /// Insert vertices into triangulation
     void insertVertices(const std::vector<V2d<T> >& vertices);
     /**
      * Insert constraints (custom-type fixed edges) into triangulation
      * @tparam TEdgeIter iterator that dereferences to custom edge type
-     * @tparam TGetEdgeVertex function object getting coordinate from* vertex.
+     * @tparam TGetEdgeVertexStart function object getting first coordinate from* vertex.
```
Comment (artem-ogre) [inline CDT/include/CDT.h:97]:
```suggestion
     * @tparam TGetEdgeVertexStart function object getting start vertex index from an edge.
```

File: CDT/include/CDT.h [hunk 4]
```diff
@@ -72,37 +72,45 @@ class CDT_EXPORT Triangulation
...
+     * @tparam TGetEdgeVertexEnd function object getting last coordinate from* vertex.
```
Comment (artem-ogre) [inline CDT/include/CDT.h:98]:
```suggestion
     * @tparam TGetEdgeVertexEnd function object getting end vertex index from an edge.
```

File: CDT/include/CDT.h [hunk 5]
```diff
@@ -200,20 +208,25 @@ struct CDT_EXPORT DuplicatesInfo
  * Find duplicates in given custom point-type range
  * @note duplicates are points with exactly same X and Y coordinates
  * @tparam TVertexIter iterator that dereferences to custom point type
- * @tparam TGetVertexCoord function object getting coordinate from* vertex.
+ * @tparam TGetVertexCoordX function object getting x coordinate from* vertex.
```
Comment (artem-ogre) [inline CDT/include/CDT.h:211]:
```suggestion
 * @tparam TGetVertexCoordX function object getting x coordinate from vertex.
```

File: CDT/include/CDT.h [hunk 6]
```diff
@@ -200,20 +208,25 @@ struct CDT_EXPORT DuplicatesInfo
...
+ * @tparam TGetVertexCoordY function object getting y coordinate from* vertex.
```
Comment (artem-ogre) [inline CDT/include/CDT.h:212]:
```suggestion
 * @tparam TGetVertexCoordY function object getting y coordinate from vertex.
```

File: CDT/include/CDT.h [hunk 7]
```diff
@@ -253,7 +266,8 @@ RemapEdges(std::vector<Edge>& edges, const std::vector<std::size_t>& mapping);
  * @note Same as a chained call of @ref FindDuplicates, @ref RemoveDuplicates,
  * and @ref RemapEdges
  * @tparam TVertexIter iterator that dereferences to custom point type
- * @tparam TGetVertexCoord function object getting coordinate from* vertex.
+ * @tparam TGetVertexCoordX function object getting x coordinate from* vertex.
```
Comment (artem-ogre) [inline CDT/include/CDT.h:269]:
```suggestion
 * @tparam TGetVertexCoordX function object getting x coordinate from vertex.
```

File: CDT/include/CDT.h [hunk 8]
```diff
@@ -253,7 +266,8 @@ RemapEdges(std::vector<Edge>& edges, const std::vector<std::size_t>& mapping);
...
+ * @tparam TGetVertexCoordY function object getting y coordinate from* vertex.
```
Comment (artem-ogre) [inline CDT/include/CDT.h:270]:
```suggestion
 * @tparam TGetVertexCoordY function object getting y coordinate from vertex.
```

**Review by artem-ogre — CHANGES_REQUESTED** [review]
Just few small doxy-fixes needed

**Comment by artem-ogre** [comment]
Thanks @tgurriet!

**Comment by tgurriet** [comment]
Thank you for this awesome library ! :)

**Pull request merged**